docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,371 @@
+Both `from` and `to` sections makes sense on the egress.
+
+| Policy                    | On sidecar    | On egress              | On both | User configurable |
+|---------------------------|---------------|------------------------|---------|-------------------|
+| MeshAccessLog             | Yes           | Yes                    | Yes     | Yes               |
+| MeshCircuitBreaker        | Yes           | Maybe                  | No      | No                |
+| MeshFaultInjection        | Maybe         | Yes                    | No      | No                |
+| MeshHealthCheck           | Yes           | Maybe                  | No      | No                |
+| MeshMetric                | Yes           | Yes                    | Yes     | Maybe             |
+| MeshProxyPatch            | Yes           | Yes                    | Yes     | Yes               |
+| MeshRateLimit             | Probably not  | Yes                    | No      | No                |
+| MeshRetry                 | Yes           | Maybe                  | No      | Maybe             |
+| MeshTimeout               | Yes           | Maybe                  | No      | No                |
+| MeshTrace                 | Yes           | Yes                    | Yes     | Maybe             |
+| MeshTrafficPermission     | Maybe         | Yes                    | Maybe   | Maybe             |
+| MeshLoadBalancingStrategy | Yes           | Yes                    | No      | Maybe             |
+| MeshTCPRoute              | Yes           | Yes                    | No      | No                |
+| MeshHTTPRoute             | Yes           | Yes                    | No      | No                |
+
+In paragraphs below I go deeper into each policy.
+
+##### MeshAccessLog
+
+MeshAccessLog is not applied on egress at all, and we don't provide a way to configure it.
+I don't see any reason not to do this though.
+
+##### MeshCircuitBreaker
+
+###### Old ExternalService behaviour
+
+Circuit Breaker is configured on the sidecar always.
+
+###### New behaviour
+
+If on egress
+- services from one zone will act as an aggregate - if a CB triggers by service-1 then service-2 will also see that result immediately
+  - this makes the configuration more sensitive to drastic changes
+  - if a MES starts failing on one endpoint /example1 that is only consumed by service-1 it will also trigger for service-2 that might be happily consuming endpoint /example2
+  - this could be mitigated by re-defining a MeshExternalService under a different name (a bit awkward)
+
+###### Verdict
+
+There might be situations where it makes sense on egress but seems like it's more natural on the sidecar.
+
+##### MeshFaultInjection
+
+###### Old ExternalService behaviour
+
+FaultInjection and MeshFaultInjection is configured for ExternalService on egress inbound.
+
+###### New behaviour
+
+If on sidecar:
+- less traffic in the cluster (it won't even exit the pod)
```

Comment:
It's a `from` policy so if supported it should be applied on the Egress no?

I thought the whole point of making Egress compulsory was to make MES a lot closer to regular services.

Reply:
> It's a from policy so if supported it should be applied on the Egress no?
> 
> I thought the whole point of making Egress compulsory was to make MES a lot closer to regular services.

It is currently an inbound policy but this could be also placed on an outbound. So while brainstorming different ideas I've put it here just so it's documented. And to me it makes sense, I don't see too many differences on putting this on the destination inbound vs source outbound. You get the same result minus network hop.

Reply:
I believe we shouldn't apply it to the outbound and should stick to `MeshFaultInjection` as an inbound policy. My argument is that you won't be able to validate if your policies (`MeshRetry`, `MeshTimeout`, etc.) work correctly because they won't be used.

Reply:
Just a reminder: this is not a proposed change, just evaluation. I added this concern as a note.